internal/controller/pkg/revision/deployment.go

```diff
@@ -104,6 +104,16 @@ func buildProviderDeployment(provider *pkgmetav1.Provider, revision v1.PackageRe
 									ContainerPort: promPortNumber,
 								},
 							},
+							Env: []corev1.EnvVar{
+								{
+									Name: "POD_NAMESPACE",
+									ValueFrom: &corev1.EnvVarSource{
+										FieldRef: &corev1.ObjectFieldSelector{
+											FieldPath: "metadata.namespace",
+										},
+									},
+								},
+							},
```

Comment:
It's probably worth a comment here explaining why we're setting `POD_NAMESPACE` for all providers, which I assume will do the same thing as `--namespace` on this PR in `init.go` (i.e. set the scope/namespace of the default `StoreConfig`).